### PR TITLE
feat(modal): Add tests for modal-compounds

### PR DESCRIPTION
**Pull request by umahaea**
- Add Tests for modal-compounds
- Fix error messaging for incorrect config object.

@JasonLantz @StommePoes @wyseguyonline @udhadpa @mhomolak for your review

### CODE REVIEW

File: src/test/java/compoundsTests/ModalTest.java [hunk 1]
```diff
@@ -0,0 +1,933 @@
...
+
+        String[] detailsPropertiesList = new String[]{"elementId", "modal-target", "componentName", "Modal"};
+        String[] propsTextList = new String[]{"initiatingButtonText", "any string", "headerTitle", "Terms n Conditions (basic title)", "closeButtonSRText", "close", "modalSaveButtonText", "save", "modalCancelButtonText", "cancel"};
+        String[] propsPropertiesList = new String[]{"isShown", "true", "id", "modalId", "cancelBtnHandler", "function () {return alert('You clicked Cancel!');}", "successBtnHandler", "function () {return alert('You clicked save!');}", "footerVisible", "true", "children", "React.createElement('p', {}, '" + longModalText + "')"};
+        setConfigAndLaunch(detailsPropertiesList, propsTextList, propsPropertiesList);
+
+        paddingTop = commonUtils.getCSSValue(modalBodyElement, "padding-top");
+        marginTop = commonUtils.getCSSValue(modalBodyElement, "margin-top");
+        marginBottom = commonUtils.getCSSValue(modalBodyElement, "margin-bottom");
+        marginTop = commonUtils.getCSSValue(modalBodyElement, "margin-top");
```
Comment (StommePoes) [inline src/test/java/compoundsTests/ModalTest.java]: margin-top twice?

File: src/test/java/compoundsTests/ModalTest.java [hunk 2]
```diff
@@ -0,0 +1,933 @@
...
+        }
+        for (int i = 0; i < paddings.length; i++) {
+            padding = commonUtils.getCSSValue(modalBodyElement, paddings[i]);
+            isPadding = commonUtils.assertValue(padding, expPaddingValue[i], "Compounds -> '" + paddings[i] + "' for " + size + " size is not as per the spec");
+            if (!isPadding) {
+                log.info("Compounds -> 'modal body' - " + paddings[i] + " for " + size + " is not as per the spec, actual: " + padding);
+            }
+            Assert.assertTrue(isPadding);
+        }
+        Assert.assertTrue(isFontSize && isLineHeight);
```
Comment (udhadpa) [inline src/test/java/compoundsTests/ModalTest.java]: Assertion for style and color is missing

File: src/test/java/compoundsTests/ModalTest.java [hunk 3]
```diff
@@ -0,0 +1,933 @@
...
+
+        String[] detailsPropertiesList = new String[]{"elementId", "modal-target", "componentName", "Modal"};
+        String[] propsTextList = new String[]{"initiatingButtonText", "any string", "headerTitle", "Terms n Conditions (basic title)", "closeButtonSRText", "close", "modalSaveButtonText", "save", "modalCancelButtonText", "cancel"};
+        String[] propsPropertiesList = new String[]{"isShown", "true", "id", "modalId", "cancelBtnHandler", "function () {return alert('You clicked Cancel!');}", "successBtnHandler", "function () {return alert('You clicked save!');}", "footerVisible", "true", "children", "React.createElement('p', {}, '" + longModalText + "')"};
+        setConfigAndLaunch(detailsPropertiesList, propsTextList, propsPropertiesList, "mobile");
+
+        paddingTop = commonUtils.getCSSValue(modalBodyElement, "padding-top", "mobile");
+        marginTop = commonUtils.getCSSValue(modalBodyElement, "margin-top", "mobile");
+        marginBottom = commonUtils.getCSSValue(modalBodyElement, "margin-bottom", "mobile");
+        marginTop = commonUtils.getCSSValue(modalBodyElement, "margin-top", "mobile");
```
Comment (StommePoes) [inline src/test/java/compoundsTests/ModalTest.java]: 2nd margin-top again

File: src/test/java/compoundsTests/ModalTest.java [hunk 4]
```diff
@@ -0,0 +1,933 @@
...
+        }
+        for (int i = 0; i < paddings.length; i++) {
+            padding = commonUtils.getCSSValue(modalBodyElement, paddings[i], "mobile");
+            isPadding = commonUtils.assertValue(padding, expPaddingValue[i], "Compounds -> 'modal body' - " + paddings[i] + " for " + size + " size is not as per the spec");
+            if (!isPadding) {
+                log.info("Compounds -> 'modal body' - " + paddings[i] + " for " + size + " is not as per the spec, actual: " + padding);
+            }
+            Assert.assertTrue(isPadding);
+        }
+        Assert.assertTrue(isFontSize && isLineHeight);
```
Comment (udhadpa) [inline src/test/java/compoundsTests/ModalTest.java]: Outline style and color assertions missing

**Comment by StommePoes** [comment]
Except for my two comments looks ok.

**Comment by udhadpa** [comment]
Rest Looks good, great work Eajaz 👍

File: src/test/java/compoundsTests/ModalTest.java (hunk 1 above)
Comment (umahaea) [inline src/test/java/compoundsTests/ModalTest.java]: oh that should be a typo, I wil fix it.

File: src/test/java/compoundsTests/ModalTest.java (hunk 2 above)
Comment (umahaea) [inline src/test/java/compoundsTests/ModalTest.java]: good catch..

File: src/test/java/elementsTests/ModalTest.java [hunk 5]
```diff
@@ -46,6 +46,9 @@ private void modalOverlayTest() {
 
         backgroundColor = commonUtils.getCSSValue(modalPgObj.modalOverlay, "background-color");
         isBackgroundColor = commonUtils.assertCSSProperties("background-color", backgroundColor, new String[]{"rgba(25, 25, 25, 0.6)", "rgb(25, 25, 25, 0.6)"});
+        if (isBackgroundColor) {
```
Comment (mhomolak) [inline src/test/java/elementsTests/ModalTest.java]: if (!isBackgroundColor) { etc

**Comment by umahaea** [comment]
got approvals, merging it.

**Pull request merged**